Ginger/Ginger/ALM/Repository/JIRA_Repository.cs

```diff
@@ -230,6 +299,119 @@ public override bool ImportSelectedTests(string importDestinationPath, IEnumerab
             return false;
         }
 
+        public bool ImportSelectedZephyrCyclesAndFolders(string importDestinationPath, IEnumerable<object> selectedObjects)
+        {
+            if (selectedObjects != null && selectedObjects.Count() > 0)
+            {
+                foreach (JiraZephyrTreeItem obj in selectedObjects)
+                {
+                    BusinessFlow existedBF;
+                    if (obj is JiraZephyrFolderTreeItem)
+                    {
+                        existedBF = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<BusinessFlow>().Where(x => x.ExternalID == ((JiraZephyrFolderTreeItem)obj).CycleId  && x.ExternalID2 == obj.Id).FirstOrDefault();
+                    }
+                    else
+                    {
+                        existedBF = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<BusinessFlow>().Where(x => x.ExternalID == obj.Id && x.ExternalID2 == null).FirstOrDefault();
+                    }
+
+                    if (existedBF != null)
+                    {
+                        Amdocs.Ginger.Common.eUserMsgSelection userSelection = Reporter.ToUser(eUserMsgKey.TestSetExists, obj.Name);
+                        if (userSelection == Amdocs.Ginger.Common.eUserMsgSelection.Yes)
+                        {
+                            File.Delete(existedBF.FileName);
+                        }
+                    }
+                    Reporter.ToStatus(eStatusMsgKey.ALMTestSetImport, null, obj.Name);
+                }
+
+                //Refresh Ginger repository and allow GingerQC to use it
+                ALMIntegration.Instance.AlmCore.GingerActivitiesGroupsRepo = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ActivitiesGroup>();
+                ALMIntegration.Instance.AlmCore.GingerActivitiesRepo = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<Activity>();
+
+                foreach (JiraZephyrTreeItem obj in selectedObjects)
+                {
+                    try
+                    {
+                        Reporter.ToStatus(eStatusMsgKey.ALMTestSetImport, null, obj.Name);
+                        JiraZephyrCycle currentCycle;
+                        BusinessFlow tsBusFlow;
+                        if (obj is JiraZephyrFolderTreeItem)
+                        {
+                            currentCycle = ((JiraCore)ALMIntegration.Instance.AlmCore).GetZephyrCycleOrFolderWithIssuesAndStepsAsCycle(obj.VersionId, ((JiraZephyrFolderTreeItem)obj).CycleId, ((JiraZephyrFolderTreeItem)obj).Id);
+                            currentCycle.name = obj.Name;
+                            currentCycle.description = ((JiraZephyrFolderTreeItem)obj).Description;
+                            tsBusFlow = ((JiraCore)ALMIntegration.Instance.AlmCore).ConvertJiraZypherCycleToBF(currentCycle);
+                            // tsBusFlow.ExternalID = ((JiraZephyrFolderTreeItem)obj).CycleId + "_|_" + obj.Id;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6280018)